discord/interactions.py

```diff
@@ -624,6 +638,94 @@ async def translate(
         return await translator.translate(string, locale=locale, context=context)
 
 
+class InteractionCallbackActivityInstance:
+    """Represents an activity instance launched as an interaction response.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`str`
+        The activity instance ID.
+    """
+
+    __slots__ = ('id',)
+
+    def __init__(self, data: InteractionCallbackActivityPayload) -> None:
+        self.id: str = data['id']
+
+
+class InteractionCallback(Generic[ClientT]):
+    """Represents an interaction response callback.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The interaction ID.
+    type: :class:`InteractionResponseType`
+        The interaction callback response type.
+    resource: Optional[Union[:class:`InteractionMessage`, :class:`InteractionCallbackActivityInstance`]]
+        The resource that the interaction response created. If a message was sent, this will be
+        a :class:`InteractionMessage`, else if an activity was launched this will be a
+        :class:`InteractionCallbackActivityInstance`. In any other case, this will be ``None``.
+    message_id: Optional[:class:`int`]
+        The message ID of the resource. Only available if the resource is a :class:`InteractionMessage`.
+    activity_id: Optional[:class:`str`]
+        The activity ID of the resource. Only available if the resource is a :class:`InteractionCallbackActivityInstance`.
+    """
+
+    def __init__(
+        self,
+        *,
+        data: InteractionCallbackPayload,
+        parent: Interaction[ClientT],
+        state: ConnectionState,
+        type: InteractionResponseType,
+    ) -> None:
+        self._state: ConnectionState = state
+        self._parent: Interaction[ClientT] = parent
+        self.type: InteractionResponseType = type
+        self._update(data)
+
+    def _update(self, data: InteractionCallbackPayload) -> None:
+        interaction = data['interaction']
+
+        self.id: int = int(interaction['id'])
+        self._thinking: bool = interaction.get('response_message_loading', False)
+        self._ephemeral: bool = interaction.get('response_message_ephemeral', False)
+
+        self.message_id: Optional[int] = utils._get_as_snowflake(interaction, 'response_message_id')
+        self.activity_id: Optional[str] = interaction.get('activity_instance_id')
+
+        self.resource: Optional[Union[InteractionMessage, InteractionCallbackActivityInstance]] = None
+
+        if 'resource' in data:
+            resource = data['resource']
```

Comment:
```suggestion
        resource = data.get('resource')
        if resource is not None:
```

Avoid doing "double checking" such as `'key' in dict` if you can help it.